Classes/Service/CompressImageService.php

```diff
@@ -382,4 +384,24 @@ protected function addMessageToFlashMessageQueue($key, array $replaceMarkers = [
         $defaultFlashMessageQueue = $flashMessageService->getMessageQueueByIdentifier();
         $defaultFlashMessageQueue->enqueue($flashMessage);
     }
+
+    /**
+     * Function to write error log and set compressed to 2 if compression failed. The file will no longer be compressed
+     * again until compressed is set ot 0 once again.
+     *
+     * @param File §file
+     * @param string $errorMessage
+     */
+    protected function excludeFile($file, $errorMessage)
+    {
+        $queryBuilder = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Database\ConnectionPool::class)->getQueryBuilderForTable('sys_file');
```

Comment:
you could use Connection directly instead of QueryBuilder

Reply:
done